### PR TITLE
BUILD(versioning): Make use of the 4th part (build/tweak) of the version

### PR DESCRIPTION
**Pull request by davidebeatrici**
The build number can be specified through the new `BUILD_NUMBER` variable.

This change is very important on Windows, for two reasons:

- It allows to easily identify binaries simply by looking at their version.
More specifically, we can guess what build they're part of.

- Right now snapshots can not be updated without uninstalling first, because the version is the same (i.e. 1.4.0).
By increasing the 4th digit of the version for every build we can solve the issue.

This pull request also takes care of renaming a few variables so that they're consistent and also clearer.

For example, `version` is now `RELEASE_ID`.

---

`BUILD_NUMBER_TOKEN` is defined on Azure Pipelines and contains a secret token that is used to access the endpoint on our server.

Restricted access is in place for the sole purpose of preventing the build number from being increased when accessing the page outside of the CI environment.

This pull request also improves the descriptive header of the build script files.

### CODE REVIEW

**Comment by davidebeatrici** [comment]
Please note that the Azure Pipelines variables are only accessible for branches that are in our repository, meaning that for pull requests like this one (whose branch is in a fork) the build number will always be 0.

I tested the code that receives the build number from our server locally and it works, but I'm not sure whether Curl is provided by default in the Windows image on Azure.

File: .ci/azure-pipelines/build_linux.bash [hunk 1]
```diff
@@ -1,26 +1,40 @@
 #!/bin/bash -ex
 #
-# Copyright 2019-2020 The Mumble Developers. All rights reserved.
+# Copyright 2019-2021 The Mumble Developers. All rights reserved.
```
Comment (Krzmbrzl) [inline .ci/azure-pipelines/build_linux.bash:3]: This is an unrelated change. We should update the copyright headers for _all files_ in a _separate commit_ :point_up:

File: .ci/azure-pipelines/build_windows.bat [hunk 2]
```diff
@@ -23,17 +23,26 @@
...
+::
+::  BUILD_NUMBER_TOKEN           - Access token for the build number page on our server.
+::
 
 @echo on
 
-for /f "tokens=* USEBACKQ" %%g in (`python "scripts/mumble-version.py"`) do (set "VER=%%g")
+if defined BUILD_NUMBER_TOKEN (
+	for /f "tokens=* USEBACKQ" %%g in (`python "scripts/mumble-version.py" --project`) do (set "VERSION=%%g")
+	for /f "tokens=* USEBACKQ" %%g in (`curl "https://mumble.info/get-build-number?version=%VERSION%&token=%BUILD_NUMBER_TOKEN%"`) do (set "BUILD_NUMBER=%%g")
```
Comment (Krzmbrzl) [inline .ci/azure-pipelines/build_windows.bat:40]: What does this do? An explanatory comment would probably be helpful here...

File: .ci/azure-pipelines/build_macos.bash [hunk 3]
```diff
@@ -1,25 +1,50 @@
...
 
-VER=$(python scripts/mumble-version.py)
+VERSION_STRING=$(python "scripts/mumble-version.py")
 
 cd $BUILD_BINARIESDIRECTORY
 
 cmake -G Ninja -DCMAKE_TOOLCHAIN_FILE=$MUMBLE_ENVIRONMENT_TOOLCHAIN -DIce_HOME="$MUMBLE_ENVIRONMENT_PATH/installed/x64-osx" \
-	-DCMAKE_BUILD_TYPE=Release -Dtests=ON -Dversion=$VER -Dstatic=ON -Dsymbols=ON -Donline-tests=ON -Dgrpc=ON \
-	-Ddisplay-install-paths=ON $BUILD_SOURCESDIRECTORY
+      -DCMAKE_BUILD_TYPE=Release -DVERSION_STRING=$VERSION_STRING -DBUILD_NUMBER=$BUILD_NUMBER \
```
Comment (Krzmbrzl) [inline .ci/azure-pipelines/build_macos.bash:45]:
```suggestion
      -DCMAKE_BUILD_TYPE=Release -DVERSION_STRING=$VERSION_STRING -Dbuild-number=$BUILD_NUMBER \
```

File: .ci/azure-pipelines/build_windows.bat [hunk 4]
```diff
@@ -51,9 +60,10 @@ del C:\Strawberry\c\bin\g++.exe
 del C:\Strawberry\c\bin\c++.exe
 
 cmake -G Ninja -DCMAKE_TOOLCHAIN_FILE="%MUMBLE_ENVIRONMENT_TOOLCHAIN%" -DVCPKG_TARGET_TRIPLET=%MUMBLE_ENVIRONMENT_TRIPLET% ^
-	-DIce_HOME="%MUMBLE_ENVIRONMENT_PATH%\installed\%MUMBLE_ENVIRONMENT_TRIPLET%" -DCMAKE_BUILD_TYPE=Release -Dtests=ON ^
-	-Dversion=%VER% -Dpackaging=ON -Dstatic=ON -Dsymbols=ON -Dasio=ON -Dg15=ON -Donline-tests=ON -Dgrpc=ON ^
-	-Ddisplay-install-paths=ON "%BUILD_SOURCESDIRECTORY%"
+      -DIce_HOME="%MUMBLE_ENVIRONMENT_PATH%\installed\%MUMBLE_ENVIRONMENT_TRIPLET%" ^
+      -DCMAKE_BUILD_TYPE=Release -DVERSION_STRING=%VERSION_STRING% -DBUILD_NUMBER=%BUILD_NUMBER% ^
```
Comment (Krzmbrzl) [inline .ci/azure-pipelines/build_windows.bat:64]:
```suggestion
      -DCMAKE_BUILD_TYPE=Release -DVERSION_STRING=%VERSION_STRING% -Dbuild-number=%BUILD_NUMBER% ^
```

File: .ci/azure-pipelines/build_linux.bash [hunk 5]
```diff
@@ -1,26 +1,40 @@
...
+fi
+
+VERSION_STRING=$(python "scripts/mumble-version.py")
 
 cd $BUILD_BINARIESDIRECTORY
 
-cmake -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=appdir/usr -Dtests=ON Donline-tests=ON -Dversion=$VER -Dsymbols=ON \
-	-Donline-tests=ON -Dgrpc=ON -Ddisplay-install-paths=ON $BUILD_SOURCESDIRECTORY
+cmake -G Ninja -DCMAKE_INSTALL_PREFIX=appdir/usr \
+      -DCMAKE_BUILD_TYPE=Release -DVERSION_STRING=$VERSION_STRING -DBUILD_NUMBER=$BUILD_NUMBER \
```
Comment (Krzmbrzl) [inline .ci/azure-pipelines/build_linux.bash:34]:
```suggestion
      -DCMAKE_BUILD_TYPE=Release -DVERSION_STRING=$VERSION_STRING -Dbuild-number=$BUILD_NUMBER \
```

File: CMakeLists.txt [hunk 6]
```diff
@@ -8,15 +8,14 @@ cmake_minimum_required(VERSION 3.15)
...
-string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]" version_short ${version})
+set(BUILD_NUMBER "0" CACHE STRING "Build number")
+set(VERSION_STRING "1.4.0" CACHE STRING "Project version (string)")
 
 # Get compilation year
 string(TIMESTAMP MUMBLE_BUILD_YEAR "%Y")
 
 project(Mumble
-	VERSION ${version_short}
+	VERSION "1.4.0.${BUILD_NUMBER}"
```
Comment (Krzmbrzl) [inline CMakeLists.txt:18]: In cases where `BUILD_NUMBER` is not defined, we get an extraneous "." at the end of our version :point_up:

File: CMakeLists.txt [hunk 7]
```diff
@@ -8,15 +8,14 @@ cmake_minimum_required(VERSION 3.15)
 cmake_policy(SET CMP0079 NEW)
 cmake_policy(SET CMP0091 NEW)
 
-set(version "1.4.0" CACHE STRING "Project version")
-
-string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]" version_short ${version})
+set(BUILD_NUMBER "0" CACHE STRING "Build number")
```
Comment (Krzmbrzl) [inline CMakeLists.txt:11]: Please overtake the variant we settled on in Nik's PR :point_up:

File: CMakeLists.txt [hunk 8]
```diff
@@ -8,15 +8,14 @@ cmake_minimum_required(VERSION 3.15)
...
+set(VERSION_STRING "1.4.0" CACHE STRING "Project version (string)")
```
Comment (Krzmbrzl) [inline CMakeLists.txt:12]: Why did you change the name of this option? I would keep it the way it was as I don't see a need to change it :thinking:

File: CMakeLists.txt [hunk 9]
```diff
@@ -62,17 +61,18 @@ endif()
...
 
 message(STATUS "##################################################")
-message(STATUS "Mumble version:  ${version}")
-message(STATUS "Architecture:    ${ARCH}")
+message(STATUS "Mumble version (string): ${VERSION_STRING}")
+message(STATUS "Mumble version:          ${PROJECT_VERSION}")
+message(STATUS "Architecture:            ${ARCH}")
 if(NOT IS_MULTI_CONFIG)
-    message(STATUS "Build type:      ${CMAKE_BUILD_TYPE}")
+    message(STATUS "Build type:              ${CMAKE_BUILD_TYPE}")
```
Comment (Krzmbrzl) [inline CMakeLists.txt:75]: What is this change about?

File: CMakeLists.txt [hunk 10]
```diff
@@ -62,17 +61,18 @@ endif()
 include(compiler)
 include(os)
 
-if (64_BIT)
+if(64_BIT)
```
Comment (Krzmbrzl) [inline CMakeLists.txt:64]: Unrelated change

File: g15helper/g15helper.rc.in [hunk 11]
```diff
@@ -20,8 +20,8 @@ IDI_ICON1 ICON DISCARDABLE "@G15HELPER_ICON@"
 #endif
 
 VS_VERSION_INFO VERSIONINFO
-	FILEVERSION		@PROJECT_VERSION_MAJOR@,@PROJECT_VERSION_MINOR@,@PROJECT_VERSION_PATCH@,0
-	PRODUCTVERSION	@PROJECT_VERSION_MAJOR@,@PROJECT_VERSION_MINOR@,@PROJECT_VERSION_PATCH@,0
+	FILEVERSION		@PROJECT_VERSION_MAJOR@,@PROJECT_VERSION_MINOR@,@PROJECT_VERSION_PATCH@,@PROJECT_VERSION_TWEAK@
```
Comment (Krzmbrzl) [inline g15helper/g15helper.rc.in:23]: Why is it called "TWEAK" here when we refer to it as "build number" seemingly everywhere else? :thinking:

File: overlay/CMakeLists.txt [hunk 12]
```diff
@@ -162,6 +162,7 @@ if(64_BIT AND MSVC)
 				"-DCMAKE_CXX_COMPILER=cl.exe"
 				"-DBUILD_OVERLAY_XCOMPILE=ON"
 				"-Dsymbols=${symbols}"
+				"-Dversion=${PROJECT_VERSION}"
```
Comment (Krzmbrzl) [inline overlay/CMakeLists.txt:165]: This does not match your changes of the variable's name

File: scripts/mumble-version.py [hunk 13]
```diff
@@ -56,6 +56,7 @@
 # 1.3.0~2020-07-02~g4f336a2~snapshot.
 
 from __future__ import (unicode_literals, print_function, division)
+from optparse import OptionParser
```
Comment (Krzmbrzl) [inline scripts/mumble-version.py:59]: The other python scripts use argparse instead. Maybe it's worth to be consistent for things like that to increase maintainability? :eyes:

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]

File: .ci/azure-pipelines/build_linux.bash (hunk 1 above)
Comment (davidebeatrici) [reply .ci/azure-pipelines/build_linux.bash:3]: I agree.

File: .ci/azure-pipelines/build_windows.bat (hunk 2 above)
Comment (davidebeatrici) [reply .ci/azure-pipelines/build_windows.bat:40]: Batch magic!

https://stackoverflow.com/a/6362922

File: CMakeLists.txt (hunk 9 above)
Comment (davidebeatrici) [reply CMakeLists.txt:75]: The message was misaligned in the CMake output.

File: CMakeLists.txt (hunk 10 above)
Comment (davidebeatrici) [reply CMakeLists.txt:64]: I think it's fine though, as I edit the part just below.

File: g15helper/g15helper.rc.in (hunk 11 above)
Comment (davidebeatrici) [reply g15helper/g15helper.rc.in:23]: https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION_TWEAK.html

File: scripts/mumble-version.py (hunk 13 above)
Comment (davidebeatrici) [reply scripts/mumble-version.py:59]: I actually took inspiration from https://github.com/mumble-voip/mumble/blob/master/macx/scripts/osxdist.py.

The signing scripts in https://github.com/mumble-voip/mumble-releng use it as well.

File: CMakeLists.txt (hunk 7 above)
Comment (davidebeatrici) [reply CMakeLists.txt:11]: `build-number`, right?

File: CMakeLists.txt (hunk 8 above)
Comment (davidebeatrici) [reply CMakeLists.txt:12]: You mean `version`? I originally changed the name because when first working on this I created another option to specify the short version.

At the end I left a single option which specifies the full version.

File: CMakeLists.txt (hunk 6 above)
Comment (davidebeatrici) [reply CMakeLists.txt:18]: It should always be defined though, as the default value is `0`.

File: .ci/azure-pipelines/build_macos.bash (hunk 3 above)
Comment (ZeroAbility) [reply .ci/azure-pipelines/build_macos.bash:45]: Since it is global, shouldn't it be capitalized, anyway? If so, I can adjust my PR after this is merged.

File: CMakeLists.txt (hunk 6 above)
Comment (ZeroAbility) [reply CMakeLists.txt:18]: > It should always be defined though, as the default value is `0`.

This didn't seem to be the case as discussed in the meeting. I would approve of that change, though. If I need to make adjustments in my PR or drop the change entirely, I can do that. I am creating the var when `packaging=ON` since there isn't really a reason to assign it otherwise.

File: CMakeLists.txt (hunk 7 above)
Comment (ZeroAbility) [reply CMakeLists.txt:11]: As discussed previously, I think the way it is represented here may be more correct.

File: .ci/azure-pipelines/build_macos.bash (hunk 3 above)
Comment (davidebeatrici) [reply .ci/azure-pipelines/build_macos.bash:45]: More specifically, it's a string instead of an `option`.

File: .ci/azure-pipelines/build_windows.bat (hunk 2 above)
Comment (Krzmbrzl) [reply .ci/azure-pipelines/build_windows.bat:40]: Holy shit is that ugly xD
We definitely need the link you just posted as a comment in the script as this is definitely something that can cause confusion ^^

File: .ci/azure-pipelines/build_macos.bash (hunk 3 above)
Comment (Krzmbrzl) [reply .ci/azure-pipelines/build_macos.bash:45]: Well it doesn't really matter if it is an option or not (imo) since options are also global variables (in a sense).
We use lowercase spelling for all other options, so why would we want to change that for the `build-number` variable?

File: CMakeLists.txt (hunk 7 above)
Comment (Krzmbrzl) [reply CMakeLists.txt:11]: > build-number, right?

Are you referring to the spelling? That's actually not what I was aiming at with my comment. I was referring to the whole cmake code that was used to introduce the variable in Nik's PR (only if `packaging=ON`, default to empty string and error if it is empty but `packaging=ON`)

File: CMakeLists.txt (hunk 8 above)
Comment (Krzmbrzl) [reply CMakeLists.txt:12]: > You mean version?

Yes I do.

> At the end I left a single option which specifies the full version.

So this change is not needed after all? In that case I'd vote for reverting it :point_up:

File: CMakeLists.txt (hunk 9 above)
Comment (Krzmbrzl) [reply CMakeLists.txt:75]: Okay. Ideally that should be a separate commit, but given it is such a minor change I guess it's fine...

File: CMakeLists.txt (hunk 10 above)
Comment (Krzmbrzl) [reply CMakeLists.txt:64]: The part just below being `set(ARCH "x64")` which remains unchanged...

In general i think we should consider using a formatter for our cmake files in order to avoid this kind of stuff in regular commits...

File: scripts/mumble-version.py (hunk 13 above)
Comment (Krzmbrzl) [reply scripts/mumble-version.py:59]: Ah okay so I guess then we are using 2 different styles already :shrug:

File: .ci/azure-pipelines/build_macos.bash (hunk 3 above)
Comment (ZeroAbility) [reply .ci/azure-pipelines/build_macos.bash:45]: I have just seen most practices with CMake where values and options that are modified from the CLI are typically all caps.

File: CMakeLists.txt (hunk 6 above)
Comment (ZeroAbility) [reply CMakeLists.txt:18]: > I have just seen most practices with CMake where values and options that are modified from the CLI are typically all caps.

So, like this `BUILD_NUMBER`. But I think that options should probably be in this syntax and variables in lower case with `-` or `_`. It would make identifying the types easier.

File: CMakeLists.txt (hunk 7 above)
Comment (ZeroAbility) [reply CMakeLists.txt:11]: > I have just seen most practices with CMake where values and options that are modified from the CLI are typically all caps.

So, like this `BUILD_NUMBER`. But I think that options should probably be in this syntax and variables in lower case with `-` or `_`. It would make identifying the types easier. Or the other way if that is more helpful.

File: .ci/azure-pipelines/build_macos.bash (hunk 3 above)
Comment (Krzmbrzl) [reply .ci/azure-pipelines/build_macos.bash:45]: But we don't follow that already. If we want to do so, I'm fine with that, but that should be done for all variables and options then and in a dedicated commit :point_up:

File: CMakeLists.txt (hunk 7 above)
Comment (Krzmbrzl) [reply CMakeLists.txt:11]: I don't think using the case of the names as a means to express the variable's type is a good thing :thinking:

File: .ci/azure-pipelines/build_windows.bat (hunk 2 above)
Comment (davidebeatrici) [reply .ci/azure-pipelines/build_windows.bat:40]: I strongly agree.

File: .ci/azure-pipelines/build_macos.bash (hunk 3 above)
Comment (davidebeatrici) [reply .ci/azure-pipelines/build_macos.bash:45]: Yeah.

File: CMakeLists.txt (hunk 6 above)
Comment (davidebeatrici) [reply CMakeLists.txt:18]: It actually does make a difference, because this PR also changes the RC templates so that the build number is shown in the binaries' (.exe/.dll) properties.

File: CMakeLists.txt (hunk 8 above)
Comment (davidebeatrici) [reply CMakeLists.txt:12]: I agree.

File: CMakeLists.txt (hunk 10 above)
Comment (davidebeatrici) [reply CMakeLists.txt:64]: I agree, https://github.com/cheshirekow/cmake_format may be a good candidate.

File: CMakeLists.txt (hunk 6 above)
Comment (Krzmbrzl) [reply CMakeLists.txt:18]: Okay in that case I'd say always default to 0 but in case `packaging=ON`, error if the variable is not set before default-initializing it. Like so:
```
if (packaging AND BUILD_NUMBER STREQUAL "")
    message(FATAL_ERROR "...")
endif()

set(BUILD_NUMBER "0" CACHE STRING "...")
```

Comment (davidebeatrici) [reply CMakeLists.txt:18]: You mean that we should begin counting at `1` instead of `0`?

File: .ci/azure-pipelines/build_windows.bat (hunk 2 above)
Comment (Krzmbrzl) [reply .ci/azure-pipelines/build_windows.bat:43]: Imo the comment should be really right there and not buried in the script's header :point_up:

File: CMakeLists.txt (hunk 7 above)
Comment (Krzmbrzl) [reply CMakeLists.txt:11]: The proper description is still missing. Could you please overtake that from Nik's PR?

File: CMakeLists.txt (hunk 6 above)
Comment (Krzmbrzl) [reply CMakeLists.txt:18]: No. But we can check whether the build number was explicitly set and enforce this to be the case if `packaging=ON` in order to avoid accidentally forgetting to set it when packaging Mumble

File: scripts/mumble-version.py (hunk 13 above)
Comment (Krzmbrzl) [reply scripts/mumble-version.py:59]: I just found https://stackoverflow.com/questions/3217673/why-use-argparse-rather-than-optparse and according to that optparse got deprecated in Python 2.7
Thus imo we should try to not use it anymore :point_up:

File: CMakeLists.txt [hunk 14]
```diff
@@ -8,15 +8,14 @@ cmake_minimum_required(VERSION 3.15)
 cmake_policy(SET CMP0079 NEW)
 cmake_policy(SET CMP0091 NEW)
 
-set(version "1.4.0" CACHE STRING "Project version")
-
-string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]" version_short ${version})
+set(BUILD_NUMBER "0" CACHE STRING "Build number")
+set(VERSION_STRING "" CACHE STRING "Project version (string)")
```
Comment (Krzmbrzl) [inline CMakeLists.txt:12]: This variable seems to be unused (other than printing it to the console)

File: src/CMakeLists.txt [hunk 15]
```diff
@@ -126,8 +126,8 @@ target_sources(shared
 
 target_compile_definitions(shared
 	PUBLIC
-		"MUMBLE_VERSION=${version}"
-		"MUMBLE_VERSION_STRING=${version_short}"
+		"MUMBLE_VERSION=${PROJECT_VERSION}"
+		$<$<NOT:$<STREQUAL:"${version}","">>:MUMBLE_VERSION_STRING=${version}>
```
Comment (Krzmbrzl) [inline src/CMakeLists.txt:130]: I think while we're on it, `MUMBLE_RELEASE_ID` would be more fitting for this define since
1. We already have `MUMBLE_VERSION` so another definition `MUMBLE_VERSION_STRING` should be expected to be the same thing, but stringified. This is not the case though
2. This doesn't necessarily have anything to do with the actual version. It's just a String identifying the version. Thus I think `ID` is the better suffix here.

Plus I'd vote for adding an explanatory comment like so:
```suggestion
		# Only define this macro, if the version variable is actually set
		$<$<NOT:$<STREQUAL:"${version}","">>:MUMBLE_RELEASE_ID=${version}>
```

File: CMakeLists.txt [hunk 16]
```diff
@@ -8,15 +8,14 @@ cmake_minimum_required(VERSION 3.15)
 cmake_policy(SET CMP0079 NEW)
 cmake_policy(SET CMP0091 NEW)
 
-set(version "1.4.0" CACHE STRING "Project version")
```
Comment (Krzmbrzl) [inline CMakeLists.txt:11]: Given that this variable is still used, I think it'd be good practice to always define it in CACHE, but default to an empty String. Furthermore I think we should update its name and description like this:
```
set(RELEASE_ID CACHE STRING "The ID of the current release (used for representation in the UI)")
```

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 14 above)
Comment (davidebeatrici) [reply CMakeLists.txt:12]: Sorry, I accidentally left it in.

File: CMakeLists.txt (hunk 6 above)
Comment (davidebeatrici) [reply CMakeLists.txt:18]: Right, I agree.

File: scripts/mumble-version.py (hunk 13 above)
Comment (davidebeatrici) [reply scripts/mumble-version.py:59]: I tested the script with Python 3.8.7.

https://docs.python.org/3/library/optparse.html

Comment (Krzmbrzl) [reply scripts/mumble-version.py:59]: I said that it seems to be _deprecated_, not that it has been removed :point_up: 
And quoting from the page you linked:

> Deprecated since version 3.2: The optparse module is deprecated and will not be developed further; development will continue with the argparse module.

Given that this is a new script, I think we should not use deprecated libraries in it :)

File: overlay/CMakeLists.txt (hunk 12 above)
Comment (Krzmbrzl) [reply overlay/CMakeLists.txt:165]: As far as I can tell the `version` parameter does no longer exist. And setting it to the project's version doesn't seem to make a lot of sense anyways...
Or is this a variable exclusive to the overlay cmake scripts?

**Review by Krzmbrzl — COMMENTED** [review]
I think we could also consider defaulting `RELEASE_ID` to `PROJECT_VERSION` at least for `packaging=ON`. That way you don't have to specify the release ID manually when building a stable package :point_up:

File: scripts/mumble-version.py (hunk 13 above)
Comment (davidebeatrici) [reply scripts/mumble-version.py:59]: Ah, sorry, I missed that part.

File: overlay/CMakeLists.txt (hunk 12 above)
Comment (davidebeatrici) [reply overlay/CMakeLists.txt:165]: https://github.com/mumble-voip/mumble/blob/fe086b41d83d6f488b5b91167c806a87094e32c0/overlay/CMakeLists.txt#L7-L18

File: .ci/azure-pipelines/build_linux.bash [hunk 17]
```diff
@@ -7,20 +7,34 @@
...
+#  BUILD_NUMBER_TOKEN      - Access token for the build number page on our server.
+#
 
-VER=$(python scripts/mumble-version.py)
+if [[ -n "$BUILD_NUMBER_TOKEN" ]]; then
+	VERSION=$(python "scripts/mumble-version.py" --project)
+	BUILD_NUMBER=$(curl "https://mumble.info/get-build-number?version=$VERSION&token=$BUILD_NUMBER_TOKEN")
+else
+	BUILD_NUMBER=0
+fi
```
Comment (Krzmbrzl) [inline .ci/azure-pipelines/build_linux.bash:27]: Actually I think we should not default to `0` in order to let cmake error about undefined build number (as `packaging=ON`). Otherwise we could accidentally always package with build number 0 which is actually against the whole point of its introduction ^^

Comment (davidebeatrici) [reply .ci/azure-pipelines/build_linux.bash:27]: It works as you described.

Comment (Krzmbrzl) [reply .ci/azure-pipelines/build_linux.bash:27]: Oh right :facepalm: 
I guess I have stared at this PR for too long now :sweat_smile:

**Review by Krzmbrzl — APPROVED** [review]
